library/core/src/char/convert.rs

```diff
@@ -155,6 +156,36 @@ impl From<char> for u128 {
     }
 }
 
+/// Map `char` with code point in U+0000..=U+00FF to byte in 0x00..=0xFF with same value, failing
+/// if the code point is greater than U+00FF.
+///
+/// Unicode is designed such that this effectively encodes bytes
```

Comment:
I don't think this should duplicate the entire documentation already present on `impl From<u8> for char`. Could you use a cross-reference instead, to just have this detailed explanation in one place?